Paintroid/src/main/java/org/catrobat/paintroid/tools/ToolType.kt

```diff
@@ -0,0 +1,198 @@
+/*
+ * Paintroid: An image manipulation application for Android.
+ * Copyright (C) 2010-2021 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.paintroid.tools
+
+import androidx.annotation.DrawableRes
+import androidx.annotation.IdRes
+import androidx.annotation.StringRes
+import org.catrobat.paintroid.R
+import org.catrobat.paintroid.common.Constants
+import org.catrobat.paintroid.tools.Tool.StateChange
+import java.util.*
+
+enum class ToolType(
+    @get:StringRes val nameResource: Int,
+    @get:StringRes val helpTextResource: Int,
+    @get:DrawableRes val drawableResource: Int,
+    private val stateChangeBehaviour: EnumSet<StateChange>,
+    @get:IdRes val toolButtonID: Int,
+    @get:DrawableRes val overlayDrawableResource: Int,
+    private val hasOptions: Boolean
+) {
+    PIPETTE(
+        R.string.button_pipette,
+        R.string.help_content_eyedropper,
+        R.drawable.ic_pocketpaint_tool_pipette,
+        EnumSet.of(StateChange.ALL),
+        1,
+        Constants.INVALID_RESOURCE_ID,
+        false
+    ),
```

Comment:
```suggestion
    PIPETTE(button_pipette, help_content_eyedropper, ic_pocketpaint_tool_pipette, EnumSet.of(StateChange.ALL), 1, INVALID_RESOURCE_ID, false),
```
If it doesn't add too many lines import these constants, please also check why toolButtonIs=1.

Reply:
toolButtinID = 1 for the pipette tool, because it signifies that that parameter doesn't have any effect there.